### PR TITLE
[KYUUBI #2035] Hive Backend Engine - `build/dist` support

### PR DESCRIPTION
**Pull request by cxzl25**
### _Why are the changes needed?_
`build/dist` supports hive.
close https://github.com/apache/incubator-kyuubi/issues/2035

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [x] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by cxzl25** [comment]
## Build
image
## List
image

**Review by yaooqinn — APPROVED** [review]

File: build/dist [hunk 1]
```diff
@@ -279,6 +295,12 @@ if [[ "$SPARK_PROVIDED" != "true" ]]; then
   cp -r "$SPARK_BUILTIN" "$DISTDIR/externals/"
 fi
 
+if [[ "$HIVE_PROVIDED" != "true" ]]; then
```
Comment (yaooqinn) [inline build/dist:298]: not related to this PR only, if we build/package without hive distribution with maven, the below will fail

Comment (cxzl25) [reply build/dist:298]: I forgot, this PR needs to be merged in https://github.com/apache/incubator-kyuubi/issues/2024 to work properly.

Comment (yaooqinn) [reply build/dist:298]: >I forgot, this PR needs to be merged in https://github.com/apache/incubator-kyuubi/issues/2024 to work properly.

yes, you are right. 

> not related to this PR only, if we build/package without hive distribution with maven, the below will fail

This is another issue we may face if we `build/dist ` with download releases skipped.

Comment (cxzl25) [reply build/dist:298]: I probably know what you mean.
If you mean to use maven -D parameter to skip the release download of hive or other engines, an error will be reported here.
```
build/dist --tgz --spark-provided --flink-provided  -Pspark-3.2 -Dhive.archive.download.skip=true
```

Comment (cxzl25) [reply build/dist:298]: Maybe do this after `#Parse arguments`.

```bash
mvn help:evaluate -Dexpression=hive.archive.download.skip -Dhive.archive.download.skip=false  | grep -v INFO | grep -v WARNING
```

Do I need to make changes in this PR? or new open issue to do this?

Comment (yaooqinn) [reply build/dist:298]: Let‘s fix this in another ticket

**Comment by cxzl25** [comment]
The following build commands are executed successfully on the local computer.

```bash
build/dist --tgz --spark-provided --flink-provided --hive-provided  -Pspark-3.2
```

```bash
build/dist --tgz --spark-provided --flink-provided   -Pspark-3.2 -Pmirror-cdn
```

**Review by ulysses-you — APPROVED** [review]

**Review by pan3793 — APPROVED** [review]

**Comment by pan3793** [comment]
Thanks, merging to master

**Pull request closed without merging**